src/player.cpp

```diff
@@ -10624,215 +10624,217 @@ bool player::wear_item(item *to_wear, bool interactive)
-        if( to_wear->covers(bp_torso) && (has_trait("SHELL") || has_trait("SHELL2")) ) {
+        if( to_wear.covers(bp_torso) && (has_trait("SHELL") || has_trait("SHELL2")) ) {
             if( interactive ) {
-                add_msg(m_info, _("You cannot fit that over your shell."));
+                add_msg_if_player(m_info, _("You cannot fit that over your shell."));
             }
             return false;
         }
 
-        if( to_wear->covers(bp_torso) &&
+        if( to_wear.covers(bp_torso) &&
             ((has_trait("INSECT_ARMS")) || (has_trait("ARACHNID_ARMS"))) ) {
             if( interactive ) {
-                add_msg(m_info, _("Your new limbs are too wriggly to fit under that."));
+                add_msg_if_player(m_info, _("Your new limbs are too wriggly to fit under that."));
             }
             return false;
         }
 
-        if( to_wear->covers(bp_head) &&
-            !to_wear->made_of("wool") && !to_wear->made_of("cotton") &&
-            !to_wear->made_of("nomex") && !to_wear->made_of("leather") &&
+        if( to_wear.covers(bp_head) &&
+            !to_wear.made_of("wool") && !to_wear.made_of("cotton") &&
+            !to_wear.made_of("nomex") && !to_wear.made_of("leather") &&
             (has_trait("HORNS_POINTED") || has_trait("ANTENNAE") || has_trait("ANTLERS"))) {
             if( interactive ) {
-                add_msg(m_info, _("You cannot wear a helmet over your %s."),
+                add_msg_if_player(m_info, _("You cannot wear a helmet over your %s."),
                            (has_trait("HORNS_POINTED") ? _("horns") :
                             (has_trait("ANTENNAE") ? _("antennae") : _("antlers"))));
             }
             return false;
         }
 
-        if (((to_wear->covers(bp_foot_l) && is_wearing_shoes("left")) ||
-              (to_wear->covers(bp_foot_r) && is_wearing_shoes("right"))) &&
-              !to_wear->has_flag("BELTED") && !to_wear->has_flag("SKINTIGHT")) {
+        if (((to_wear.covers(bp_foot_l) && is_wearing_shoes("left")) ||
+              (to_wear.covers(bp_foot_r) && is_wearing_shoes("right"))) &&
+              !to_wear.has_flag("BELTED") && !to_wear.has_flag("SKINTIGHT")) {
             // Checks to see if the player is wearing shoes
             if(interactive){
-                add_msg(m_info, _("You're already wearing footwear!"));
+                add_msg_if_player(m_info, _("You're already wearing footwear!"));
             }
             return false;
         }
     }
 
-    if (to_wear->invlet == 0) {
-        inv.assign_empty_invlet( *to_wear, false );
-    }
-
     const bool was_deaf = is_deaf();
-    last_item = itype_id(to_wear->type->id);
-    worn.push_back(*to_wear);
+    last_item = itype_id(to_wear.type->id);
+    worn.push_back(to_wear);
 
-    if(interactive)
-    {
-        add_msg(_("You put on your %s."), to_wear->tname().c_str());
+    if( interactive ) {
+        add_msg( _("You put on your %s."), to_wear.tname().c_str() );
```

Comment:
This still calls `add_msg`, is this intended? (Because all the other calls to it have been changed to `add_msg_if_player`.)